function-aws-api-proxy/src/test/groovy/io/micronaut/function/aws/proxy/MapCollapseUtilsSpec.groovy

```diff
@@ -0,0 +1,77 @@
+package io.micronaut.function.aws.proxy
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2HTTPEvent
+import io.micronaut.core.convert.ConversionService
+import io.micronaut.http.CaseInsensitiveMutableHttpHeaders
+import io.micronaut.http.HttpHeaders
+import io.micronaut.http.MutableHttpHeaders
+import io.micronaut.test.extensions.spock.annotation.MicronautTest
+import jakarta.inject.Inject
+import spock.lang.Specification
+
+@MicronautTest
+class MapCollapseUtilsSpec extends Specification {
```

Comment:
The examples `Foo` and `Key` in this test are taken directly from the AWS Lambda Console. I did request and logged them to see how they are represented in the Lambda Event Objects.